project/Build.scala

```diff
@@ -228,7 +228,15 @@ object Build {
 
 
   // Bootstrap with -optimise
-  lazy val commonOptimisedSettings = commonBootstrappedSettings ++ Seq(scalacOptions ++= Seq("-optimise"))
+  lazy val commonOptimisedSettings = commonBootstrappedSettings ++ Seq(
+    scalacOptions ++= Seq("-optimise"),
+
+    // The *-bootstrapped and *-optimised projects contain the same sources, so
+    // we only need to import one set in the IDE. We prefer to import the
+    // non-optimized projects because optimize is slower to compile and we do
+    // not trust its output yet.
+    excludeFromIDE := true
```

Comment:
I read the above comment as "`*-bootstrapped` projects must be skipped, so why isn't this in `commonBootstrappedSettings` instead? My guess is that it's because `dotty-language-server` exists only in bootstrapped-mode and shouldn't be skipped?

Interestingly, `commonBootstrappedSettings` contains: 
```scala
EclipseKeys.skipProject := true
```

It would probably be less surprising if kept close to `excludeFromIDE := true`.

Reply:
There are three kinds of projects with the same set of sources:
- Non-bootstrapped (compiled with Scala 2.12)
- Bootstrapped (compiled with Dotty non-bootstrapped, default options)
- Optimized (compiled with Dotty non-bootstrapped, with -optimise added)

For the Dotty IDE, the non-bootstrapped don't matter: they're not Dotty-compiled projects so they will be skipped by the plugin, so we only need to decide if we want the bootstrapped or the optimized one, we prefer the bootstrapped non-optimized for the reason given by the comment.

For Eclipse, the situation is different: the only projects which we want to import are the non-bootstrapped one, because the Scala IDE can only deal with Scala 2-compiled projects, so we skip everything else (commonBootstrappedSettings is also part of the settings used for optimized projects).

Hope this clears things up :).